proxy/http2/Http2ClientSession.h

```diff
@@ -318,3 +320,9 @@ Http2ClientSession::write_buffer_size()
 {
   return write_buffer->max_read_avail();
 }
+
+inline bool
+Http2ClientSession::is_tls() const
+{
+  return true;
```

Comment:
Did we deprecate HTTP/2 over TCP? IIRC, We don't support "upgrade" from HTTP/1.1 to HTTP/2, but
clients can start a HTTP/2 over TCP with HTTP/2 Connection Preface directly.

Reply:
Does that work?  If so, we should probably add a test case for it because I'd be concerned that it would continue to work.